### PR TITLE
102300404 - work2

### PR DESCRIPTION
**Pull request by Thinkweavey**

### CODE REVIEW

**Comment by Thinkweavey** [comment]
第二轮考核作业

File: work2/143885431/AnimalNotFountException.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+public class AnimalNotFountException extends RuntimeException {
+    public AnimalNotFountException() {
+    }
+
+    public AnimalNotFountException(String msg) {
+        super(msg);
+    }
+
+    public String toString() {
```
Comment (Kikikisum) [inline work2/143885431/AnimalNotFountException.java:14]: 覆写

File: work2/143885431/Customer.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import java.time.LocalDate;
+
+public class Customer {
+    String name;
+    int times;
+    LocalDate time;
```
Comment (Kikikisum) [inline work2/143885431/Customer.java:6]: 访问权限?

File: work2/143885431/MyAnimalShop.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * @author think
+ */
+public class MyAnimalShop implements AnimalShop{
+    private double balance;
+    ArrayList<Animal> animalArrayList = new ArrayList<>();
```
Comment (Kikikisum) [inline work2/143885431/MyAnimalShop.java:10]: 访问权限?

File: work2/143885431/MyAnimalShop.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    @Override
+    public void buy(Animal animal,int num) {
+        double buySum = 0;
+        try {
+            buySum = (animal.getPrice()-55) * num;
+            if (buySum > this.balance) {
+                throw new InsufficientBalanceException("余额不足");
+            }
+        } catch (InsufficientBalanceException e) {
+            System.out.println(e.toString());
```
Comment (Kikikisum) [inline work2/143885431/MyAnimalShop.java:69]: 不要再用toString函数了

File: work2/143885431/MyAnimalShop.java [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+            for (int i = 0; i < num; i++) {
+                animalArrayList.add(animal);
+            }
+            System.out.println("购买成功,购买了" + num + "只" + animal);
+        }
+        if (num<0) {
+            System.out.println("购买失败，购买数量应大于0");
+        }
+    }
+    public void sell(Animal animal,int num){
```
Comment (Kikikisum) [inline work2/143885431/MyAnimalShop.java:82]: 你的num代表的是什么，如果是购买动物的个数，为什么可以通过total和num的比较得出了“店内没有动物”？后面的就会出现动物不够买卖的情况

File: work2/143885431/MyAnimalShop.java [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+
+    public void reception(Customer customer) {
+        for (int i = 0; i < customerArrayList.size(); i++) {
+            if(customerArrayList.get(i).getName().equals(customer.getName())) {
+                System.out.println("欢迎再次光临");
+            }else {
+                customerArrayList.add(customer);
+                System.out.println("欢迎光临");
+                customer.setTimes(customer.getTimes()+1);
```
Comment (Kikikisum) [inline work2/143885431/MyAnimalShop.java:110]: 只有第一次来的顾客需要更新到店次数？是不是放错位置了

File: work2/143885431/MyAnimalShop.java [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+        int a=1;
+        for (int i = 0; i < animalArrayList.size(); i++) {
+            if(animalArrayList.get(i).getName().equals(animal)){
+                a=0;
+                System.out.println("好的，谢谢惠顾！");
+                animalArrayList.remove(i);
+            }
+        }
+        if(a==1){
+            System.out.println("很抱歉，我们这里没有这只宠物.");
```
Comment (Kikikisum) [inline work2/143885431/MyAnimalShop.java:129]: 异常处理？

File: work2/143885431/MyAnimalShop.java [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+        for (Animal animal1 : animalArrayList) {
+            System.out.println(animal1);
+        }
+        System.out.println(customer.getName() + "先生（女士），请问您想要哪只宠物？");
+        Scanner scanner1 = new Scanner(System.in);
+        String animal =scanner1.nextLine();
+        int a=1;
+        for (int i = 0; i < animalArrayList.size(); i++) {
+            if(animalArrayList.get(i).getName().equals(animal)){
+                a=0;
```
Comment (Kikikisum) [inline work2/143885431/MyAnimalShop.java:123]: 购买完要变化宠物店的余额

**Comment by Thinkweavey** [comment]
跪谢大佬！！！



---Original---
From: ***@***.***&gt;
Date: Tue, Nov 7, 2023 10:50 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [west2-online-reserve/collection-java] 102300404 - work2 (PR#202)




 
@Kikikisum commented on this pull request.
 
 
In work2/143885431/AnimalNotFountException.java:
 &gt; @@ -0,0 +1,17 @@ +// +// Source code recreated from a .class file by IntelliJ IDEA +// (powered by FernFlower decompiler) +// + +public class AnimalNotFountException extends RuntimeException { +    public AnimalNotFountException() { +    } + +    public AnimalNotFountException(String msg) { +        super(msg); +    } + +    public String toString() {  
覆写
 
 
In work2/143885431/Customer.java:
 &gt; @@ -0,0 +1,47 @@ +import java.time.LocalDate; + +public class Customer { +    String name; +    int times; +    LocalDate time;  
访问权限?
 
 
In work2/143885431/MyAnimalShop.java:
 &gt; @@ -0,0 +1,150 @@ +import java.time.LocalDate; +import java.util.ArrayList; +import java.util.Scanner; + +/** + * @author think + */ +public class MyAnimalShop implements AnimalShop{ +    private double balance; +    ArrayList<Animal&gt; animalArrayList = new ArrayList<&gt;();  
访问权限?
 
 
In work2/143885431/MyAnimalShop.java:
 &gt; + +    public void setTotal(int total) { +        this.total = total; +    } + + +    @Override +    public void buy(Animal animal,int num) { +        double buySum = 0; +        try { +            buySum = (animal.getPrice()-55) * num; +            if (buySum &gt; this.balance) { +                throw new InsufficientBalanceException("余额不足"); +            } +        } catch (InsufficientBalanceException e) { +            System.out.println(e.toString());  
不要再用toString函数了
 
 
In work2/143885431/MyAnimalShop.java:
 &gt; +            } +        } catch (InsufficientBalanceException e) { +            System.out.println(e.toString()); +        } +        if (num &gt; 0&amp;&amp; buySum <this.balance) { +            this.balance = this.balance - buySum; +            for (int i = 0; i < num; i++) { +                animalArrayList.add(animal); +            } +            System.out.println("购买成功,购买了" + num + "只" + animal); +        } +        if (num<0) { +            System.out.println("购买失败，购买数量应大于0"); +        } +    } +    public void sell(Animal animal,int num){  
你的num代表的是什么，如果是购买动物的个数，为什么可以通过total和num的比较得出了“店内没有动物”？后面的就会出现动物不够买卖的情况
 
 
In work2/143885431/MyAnimalShop.java:
 &gt; +                    animalArrayList.remove(i); +                } +            } +            System.out.println("卖出了" + num + "只" + animal.toString()); +        } +    } + + +    public void reception(Customer customer) { +        for (int i = 0; i < customerArrayList.size(); i++) { +            if(customerArrayList.get(i).getName().equals(customer.getName())) { +                System.out.println("欢迎再次光临"); +            }else { +                customerArrayList.add(customer); +                System.out.println("欢迎光临"); +                customer.setTimes(customer.getTimes()+1);  
只有第一次来的顾客需要更新到店次数？是不是放错位置了
 
 
In work2/143885431/MyAnimalShop.java:
 &gt; +        for (Animal animal1 : animalArrayList) { +            System.out.println(animal1); +        } +        System.out.println(customer.getName() + "先生（女士），请问您想要哪只宠物？"); +        Scanner scanner1 = new Scanner(System.in); +        String animal =scanner1.nextLine(); +        int a=1; +        for (int i = 0; i < animalArrayList.size(); i++) { +            if(animalArrayList.get(i).getName().equals(animal)){ +                a=0; +                System.out.println("好的，谢谢惠顾！"); +                animalArrayList.remove(i); +            } +        } +        if(a==1){ +            System.out.println("很抱歉，我们这里没有这只宠物.");  
异常处理？
 
 
In work2/143885431/MyAnimalShop.java:
 &gt; +                customerArrayList.add(customer); +                System.out.println("欢迎光临"); +                customer.setTimes(customer.getTimes()+1); +            } +        } +        System.out.println("我们这里有："); +        for (Animal animal1 : animalArrayList) { +            System.out.println(animal1); +        } +        System.out.println(customer.getName() + "先生（女士），请问您想要哪只宠物？"); +        Scanner scanner1 = new Scanner(System.in); +        String animal =scanner1.nextLine(); +        int a=1; +        for (int i = 0; i < animalArrayList.size(); i++) { +            if(animalArrayList.get(i).getName().equals(animal)){ +                a=0;  
购买完要变化宠物店的余额

**Pull request closed without merging**